metadata_xml/validation.py

```diff
@@ -183,6 +249,19 @@ def validate(record):
     if fields_with_bad_dates:
         errors.append("""Date/time formatting error in field(s): {}.
                                  Pattern must match one of: {}""".format(
-            join(fields_with_bad_dates), join(acceptable_date_formats_text)))
+            join(fields_with_bad_dates), join(acceptable_date_formats)))
+
+    if fields_with_bad_emails:
+        errors.append(
+            f"Bad email address in field(s): {fields_with_bad_emails}")
+
+    if fields_with_nonnumeric:
+        errors.append(
+            f"Non-numeric values found in field(s): {fields_with_nonnumeric}")
+
+    if fields_with_bad_durations:
+        errors.append(
+            f"""Durations dont conform to ISO 8601 duration format:
+                {fields_with_bad_durations}""")
```

Comment:
Seems bit odd to return a list of error strings rather than actually throwing the errors?

Reply:
I did it this way because execution has to continue, as there can be multiple errors for one dataset. Though there could be a pattern that catches exceptions as well, something to look into. 

I also think it would be nice to switch to a real validation library at some point, eg see a list of them here https://github.com/mahmoudimus/awesome-validation-python